### PR TITLE
DM-34491: DM-27013 broke ap_verify runs

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: config/calibrate.py [hunk 1]
```diff
@@ -2,26 +2,16 @@
...
-# band with the most depth)
-# Use panstarrs for photometry (grizy filters)
-for refObjLoader in (config.astromRefObjLoader,
-                     config.photoRefObjLoader,):
-    refObjLoader.retarget(LoadIndexedReferenceObjectsTask)
-    # HSC has *lots* of filters, easier to modify the official config
-    refObjLoader.load(os.path.join(hscDir, "filterMap.py"))
-    # Official config doesn't bother mapping ugrizy names, which are used in Gen 2
-    for hscFilter in ["u", "g", "r", "i", "z", "y"]:
-        config.astromRefObjLoader.filterMap[hscFilter] = hscFilter
```
Comment (kfindeisen) [inline config/calibrate.py:19]: Can you explain why you removed this? All the dataset configs explicitly set the `refObjLoader` tasks so that we're configuring the task we expect to be, even if the default or the obs config changes out from under us. The fact that at least two completely different changes are being blended (and "and"ed) in the same commit doesn't help.

File: config/calibrate.py [hunk 2]
```diff
@@ -2,26 +2,16 @@
...
-
+# band with the most depth).
 config.connections.astromRefCat = "gaia"
 config.astromRefObjLoader.ref_dataset_name = config.connections.astromRefCat
-for hscFilter in config.astromRefObjLoader.filterMap:
-    config.astromRefObjLoader.filterMap[hscFilter] = "phot_g_mean"
+config.astromRefObjLoader.anyFilterMapsToThis = "phot_g_mean"
+config.astromRefObjLoader.filterMap = {}
```
Comment (kfindeisen) [inline config/calibrate.py:13]: Why can't this change be its own commit? It should be sufficient to fix the actual bug.

**Review by kfindeisen — COMMENTED** [review]
I'd like some clarification on the nature of this "and" commit; I only understand half of the changes being made.

File: config/calibrate.py (hunk 1 above)
Comment (parejkoj) [reply config/calibrate.py:19]: `LoadIndexedReferenceObjectsTask` is a gen2-only thing, and I had to unravel the `for` loop because the astrometry and photometry refcats use different configs (no filterMap for Gaia and none needed for ps1).

File: config/calibrate.py (hunk 2 above)
Comment (parejkoj) [reply config/calibrate.py:13]: It's not enough, because the `for` loop above sets a filterMap for astrometry that shouldn't exist.

Comment (kfindeisen) [reply config/calibrate.py:13]: And new line 13 overwrites it...

File: config/calibrate.py (hunk 1 above)
Comment (kfindeisen) [reply config/calibrate.py:19]: Ah, [so I misread the Gen 3 code](https://github.com/lsst/pipe_tasks/blob/main/python/lsst/pipe/tasks/calibrate.py#L636-L640). Would it be useful to explicitly retarget to `ReferenceObjectLoader`, for the same reasons as above?

In any case, if you're changing this part of the config, you should also change it for the HiTS datasets.

Comment (parejkoj) [reply config/calibrate.py:19]: ReferenceObjectLoader isn't a Task, it happens during QG generation, so there's nothing to retarget.

Good point on changing the HITS datasets, too.
